pkg/event-manager/drivers/k8s_backend.go

```diff
@@ -237,10 +271,40 @@ func (k *k8sBackend) Delete(ctx context.Context, driver *entities.Driver) error
 	return nil
 }
 
-func (k *k8sBackend) Update(ctx context.Context, driver *entities.Driver) error {
-	span, ctx := trace.Trace(ctx, "")
-	defer span.Finish()
+// TODO: find better way of getting pod failure reason
+func getReasonForUnavailablePods(pods *corev1.PodList, fullname string) error {
+	for _, pod := range pods.Items {
+		if !strings.Contains(pod.Name, fullname) {
+			continue
+		}
+
+		for _, containerStatus := range pod.Status.ContainerStatuses {
+			if containerStatus.Name != "driver" {
+				continue
+			}
+
+			if containerStatus.State.Waiting != nil {
+				waiting := containerStatus.State.Waiting
+				return &EventdriverErrorDeploymentNotAvaialble{
+					Err: ewrapper.Errorf("k8s: deployment `%s` is `waiting`: %s, %s",
+						fullname, waiting.Reason, waiting.Message),
+				}
+			}
+			if containerStatus.State.Terminated != nil {
+				terminated := containerStatus.State.Terminated
+				return &EventdriverErrorDeploymentNotAvaialble{
+					Err: ewrapper.Errorf("k8s: deployment `%s` is `terminated`: %s, %s",
+						fullname, terminated.Reason, terminated.Message),
+				}
+			}
 
+		}
+	}
+
+	return nil
+}
+
+func (k *k8sBackend) Update(ctx context.Context, driver *entities.Driver) error {
```

Comment:
Make sure to add a comment on all public methods (we need to keep running lint)

Reply:
Yeah, sure, i will make sure and double check it. For this case, golint wasn't complaining since `k8sBackend` was treated as unexported, it's methods were treated as unexported too.